internal/resources/kubeadm_utils.go

```diff
@@ -63,7 +63,7 @@ func GetKubeadmManifestDeps(ctx context.Context, client client.Client, tenantCon
 		if len(kubeProxy.ImageRepository) > 0 {
 			config.Parameters.KubeProxyOptions.Repository = kubeProxy.ImageRepository
 		} else {
-			config.Parameters.KubeProxyOptions.Repository = "k8s.gcr.io"
+			config.Parameters.KubeProxyOptions.Repository = "registry.k8s.io"
```

Comment:
May I ask you to test locally? Just wanted to be sure `kube-proxy` is already published there.

You know, famous last words.

Reply:
```
k8s.gcr.io/kube-apiserver                                           v1.24.3              d521dd763e2e   7 months ago    130MB
registry.k8s.io/kube-apiserver                                      v1.24.3              d521dd763e2e   7 months ago    130MB
k8s.gcr.io/kube-controller-manager                                  v1.24.3              586c112956df   7 months ago    119MB
registry.k8s.io/kube-controller-manager                             v1.24.3              586c112956df   7 months ago    119MB
k8s.gcr.io/kube-scheduler                                           v1.24.3              3a5aa3a515f5   7 months ago    51MB
registry.k8s.io/kube-scheduler                                      v1.24.3              3a5aa3a515f5   7 months ago    51MB
registry.k8s.io/kube-proxy                                          v1.24.3              2ae1ba6417cb   7 months ago    110MB
k8s.gcr.io/kube-proxy                                               v1.24.3              2ae1ba6417cb   7 months ago    110MB
```

LGTM